pkg/ring/kv/memberlist/memberlist_logger.go

```diff
@@ -0,0 +1,108 @@
+package memberlist
+
+import (
+	"io"
+	"regexp"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+// StdlibAdapter wraps a Logger and allows it to be passed to the stdlib
+// logger's SetOutput. It will extract date/timestamps, filenames, and
+// messages, and place them under relevant keys.
+type loggerAdapter struct {
```

Comment:
This is a nifty little struct. It seems more tied to the stdlib and not memberlist. Maybe move this to the utils package.

Reply:
This is in fact copy of https://godoc.org/github.com/go-kit/kit/log#StdlibAdapter, with modifications to understand output produced by memberlist library (esp. log level). See usage of m.logger here: https://github.com/hashicorp/memberlist/blob/master/memberlist.go

I've updated the comment to reflect that.

Reply:
Cool